File: src/java/org/elastos/hive/subscription/payment/CreateOrderParams.java
package org.elastos.hive.subscription.payment;

import com.google.gson.annotations.SerializedName;

class CreateOrderParams {
	@SerializedName("subscription")
	private String subscription;

	@SerializedName("pricing_name")
	private String pricingName;

	public CreateOrderParams(String subscription, String pricingName) {
		this.subscription = subscription;
		this.pricingName = pricingName;
	}
}

File: src/domain/payment/createorderparams.ts
export class CreateOrderParams {
//	@SerializedName("subscription")
	private subscription: string;

//	@SerializedName("pricing_name")
	private pricingName: string;

    public getSubscription(): string {
        return this.subscription;
    }

    public getPricingName(): string {
        return this.pricingName;
    }

    public setSubscription(subscription: string): CreateOrderParams {
        this.subscription = subscription;
        return this;
    }

    public setPricingName(pricingName: string): CreateOrderParams {
        this.pricingName = pricingName;
        return this;
    }
}
